modules/compliance-tests/src/smithy4s/compliancetests/internals/ServerHttpComplianceTestCase.scala

```diff
@@ -66,13 +68,33 @@ private[compliancetests] class ServerHttpComplianceTestCase[
       .fromString(testCase.method)
       .getOrElse(sys.error("Invalid method"))
 
+    val expectedQueryParams: Vector[(String, Option[String])] =
+      parseQueryParams(testCase.queryParams)
+        .foldLeft[ListMap[String, Vector[Option[String]]]](ListMap.empty) {
+          case (acc, (k, v)) =>
+            acc.get(k) match {
+              case Some(value) => acc + (k -> (value :+ v))
+              case None        => acc + (k -> Vector(v))
+            }
+        }
+        .map {
+          // FIXME: replacing single query parameter without value with empty string
+          // to make sure that https://github.com/smithy-lang/smithy/blob/6c42bc9d60a681e63c66f4cde33d7a189a1ff9a6/smithy-aws-protocol-tests/model/restJson1/http-query.smithy#L476-L489
+          // passes.
+          // Previously this was done in `parseQueryParams`.
+          case (k, Vector(None)) => k -> Vector(Some(""))
+          case kv                => kv
+        }
+        .toVector
+        .flatMap { case (k, v) =>
+          v.map(k -> _)
+        }
+
```

Comment:
This is a bit of a hack to get the compliance tests passing, but will require some discussion.

As far as I can tell, this [aws restJson compliance test](https://github.com/smithy-lang/smithy/blob/6c42bc9d60a681e63c66f4cde33d7a189a1ff9a6/smithy-aws-protocol-tests/model/restJson1/http-query.smithy#L476-L489) expects a query of `example.com/?foo` to be treated as `foo=Some("")` instead of `foo=None`.